hdkeychain/extendedkey_test.go

```diff
@@ -610,6 +616,20 @@ func TestExtendedKeyAPI(t *testing.T) {
 			continue
 		}
 
+		chainCode := key.ChainCode()
+		if !bytes.Equal(chainCode, test.chainCode) {
+			t.Errorf("ChainCode #%d (%s): want %v, got %v", i,
+				test.name, chainCode, test.chainCode)
+			continue
+		}
+
+		childNum := key.ChildNum()
+		if childNum != test.childNum {
+			t.Errorf("ChildNum #%d (%s): want %d, got %d", i,
+				test.name, childNum, test.childNum)
```

Comment:
It seems the "want" and "got" are flipped.

```suggestion
				test.name, test.childNum, childNum)
```